data/systemd/snapd.manage

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+#
+# This script acts as an entry point for running snapd from snapd.snap on
+# a base snap that otherwise doesn't ship snapd itself. The entry point
+# arranges for appropriate system integration, in particular by creating
+# and starting systemd units for running snapd without any confinement.
+#
+# Arguments:
+# $1: start|stop
+# $2: prefix of the installed snapd snap (e.g. /snap/snapd/42/)
+#     (on a fedora base this might be e.g. /var/lib/snaps...)
+
+set -e
+
+CMD="$1"
+PREFIX="$2"
+
+UNITS="snapd.service snapd.autoimport.service snapd.core-fixup.service \
+       snapd.seeded.service snapd.snap-repair.service \
+       snapd.snap-repair.timer snapd.socket"
+ENABLE_UNITS="snapd.system-shutdown.service"
+
+# TODO: discuss ID_LIKE=ubuntu-core or ID_LIKE=snap-core ?
+if ! grep -q -E "ID(_LIKE)?=ubuntu-core" /etc/os-release && \
+        ! grep -q -E "ID(_LIKE)?=ubuntu-core" /usr/lib/os-release;
+    then
+        echo "cannot run this helper on a non ubuntu-core system"
+    exit 1
+fi
+
+case $CMD in
+    start)
+        echo "Running start action"
+        if [ "$PREFIX" = "" ]; then
+            echo "cannot run without the prefix for the snapd snap"
+            exit 1
+        fi
+        # use existing snapd.service as a template and modify so that the
+        # right prefix is used
+        for u in $UNITS $ENABLE_UNITS; do
+            sed "s#ExecStart=#ExecStart=$PREFIX#" "$PREFIX/lib/systemd/system/$u" > "/run/systemd/system/$u"
+        done
+        # and now use systemd to start it
+        systemctl daemon-reload
+        # some units only need to get enabled
+        for unit in $ENABLE_UNITS; do
+            if ! systemctl is-enabled $unit; then
```

Comment:
This doesn't feel shellcheck clean. Is shellcheck really working?

Reply:
Just run it though shellcheck locally. I think this is similar to what was discussed in one of my shellcheck PRs. AFAICT shellcheck knows the value of ENABLE_UNITS and UNITS variables and it can prove that there is no unintentional word splitting.

Reply:
Ahh, nice static analysis. Cool

Reply:
Well, if I change e.g. 
```
-            sed "s#ExecStart=#ExecStart=$PREFIX#" "$PREFIX/lib/systemd/system/$u" > "/run/systemd/system/$u"
+            sed "s#ExecStart=#ExecStart=$PREFIX#" "$PREFIX/lib/systemd/system/$u" > /run/systemd/system/$u
```
it will complain:
```
In snapd.manage line 41:
            sed "s#ExecStart=#ExecStart=$PREFIX#" "$PREFIX/lib/systemd/system/$u" > /run/systemd/system/$u
                                                                                                        ^-- SC2086: Double quote to prevent globbing and word splitting.
```
So it is working. I suspect it has some smarts because ENABLE_UNITS is statically defined?